File: core/src/net/natruid/jungle/components/UnitComponent.java
package net.natruid.jungle.components;

import com.artemis.Component;
import com.artemis.annotations.EntityId;
import com.artemis.annotations.PooledWeaver;
import com.badlogic.gdx.utils.Array;
import com.badlogic.gdx.utils.ObjectMap;
import net.natruid.jungle.utils.Faction;
import net.natruid.jungle.utils.skill.Skill;

@PooledWeaver
public class UnitComponent extends Component {
    public final Array<Skill> skills = new Array<>();
    public final ObjectMap<String, Integer> proficiencies = new ObjectMap<>();
    @EntityId
    public int tile = -1;
    public Faction faction = Faction.NONE;
    public int level = 0;
    public int exp = 0;
    public int hp = 0;
    public int ap = 0;
    public float extraMovement = 0;

    protected void reset() {
        faction = Faction.NONE;
        skills.clear();
        proficiencies.clear();
    }
}

File: core/src/net/natruid/jungle/components/UnitComponent.kt
package net.natruid.jungle.components

import com.artemis.PooledComponent
import com.artemis.annotations.EntityId
import net.natruid.jungle.utils.Faction
import net.natruid.jungle.utils.skill.Skill

class UnitComponent : PooledComponent() {
    @EntityId
    var tile: Int = -1
    var faction: Faction = Faction.NONE
    var level: Int = 0
    var exp: Int = 0
    var hp: Int = 0
    var ap: Int = 0
    var extraMovement: Float = 0f
    var hasTurn: Boolean = false
    val skills: ArrayList<Skill> = ArrayList()

    override fun reset() {
        tile = -1
        faction = Faction.NONE
        level = 0
        exp = 0
        hp = 0
        ap = 0
        extraMovement = 0f
        hasTurn = false
        skills.clear()
    }
}
